codelists/forms.py

```diff
@@ -56,7 +89,7 @@ def clean_csv_data(self):
         return data
 
 
-class CodelistCreateForm(forms.ModelForm, CSVValidationMixin):
+class CodelistCreateForm(CodelistUniquenessMixin, forms.ModelForm, CSVValidationMixin):
```

Comment:
Interesting choice of sorting mixins alphabetically, by lower-casing the class names and stripping the first letter...

Reply:
Hah!  Not quite though…

Superclass/mixin order is important here unfortunately.  CodelistUniquenessMixin's `full_clean()` needs to be the direct parent of `CodelistCreateForm` so _it's_ call to `super()` calls `ModelForm` to populate the instance values from the Form values so it can then handle the uniqueness check for us.

Reply:
Oof.